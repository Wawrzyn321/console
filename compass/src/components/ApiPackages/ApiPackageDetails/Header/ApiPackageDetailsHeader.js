import React from 'react';
import LuigiClient from '@kyma-project/luigi-client';
import PropTypes from 'prop-types';

import { PageHeader, handleDelete } from 'react-shared';
import { Button } from 'fundamental-react';
import RequestInputSchemaModal from '../RequestInputSchemaModal/RequestInputSchemaModal';

import { useMutation } from '@apollo/react-hooks';
import {
  GET_API_PACKAGE,
  DELETE_API_PACKAGE,
} from 'components/ApiPackages/gql';
import { SEND_NOTIFICATION } from 'gql';

ApiPackageDetailsHeader.propTypes = {
  apiPackage: PropTypes.object.isRequired,
  application: PropTypes.object.isRequired,
};

function navigateToApplication() {
  LuigiClient.linkManager()
    .fromContext('application')
    .navigate('');
}

export default function ApiPackageDetailsHeader({ apiPackage, application }) {
  const [sendNotification] = useMutation(SEND_NOTIFICATION);

  const [deleteApiPackageMutation] = useMutation(DELETE_API_PACKAGE, {
    refetchQueries: () => [
      {
        query: GET_API_PACKAGE,
        variables: {
          applicationId: application.id,
          apiPackageId: apiPackage.id,
        },
      },
    ],
  });

  function showDeleteSuccessNotification() {
    sendNotification({
      variables: {
        content: `Deleted API Package "${apiPackage.name}".`,
        title: apiPackage.name,
        color: '#359c46',
        icon: 'accept',
        instanceName: apiPackage.name,
      },
    });
  }

  const breadcrumbItems = [
    { name: 'Applications', path: '/applications', fromContext: 'tenant' },
    { name: application.name, path: '/', fromContext: 'application' },
    {
      name: apiPackage.name,
      path: '/',
    },
    { name: '' },
  ];

  const deleteApiPackage = () => {
    handleDelete(
      'API Package',
      apiPackage.id,
      apiPackage.name,
      id => deleteApiPackageMutation({ variables: { id } }),
      () => {
        navigateToApplication();
        showDeleteSuccessNotification();
      },
    );
  };

  const deleteButton = (
    <Button onClick={deleteApiPackage} type="negative" option="light">
      Delete
    </Button>
  );

  return (
    <PageHeader
      title={apiPackage.name}
      breadcrumbItems={breadcrumbItems}
      actions={deleteButton}
    >
      <PageHeader.Column title="Name">{apiPackage.name}</PageHeader.Column>
      <PageHeader.Column title="Description" columnSpan={2}>
        {apiPackage.description}
      </PageHeader.Column>
      <PageHeader.Column title="Request Input Schema" columnSpan={3}>
        <RequestInputSchemaModal
          schema={apiPackage.instanceAuthRequestInputSchema}
        />
      </PageHeader.Column>
    </PageHeader>
  );
}
