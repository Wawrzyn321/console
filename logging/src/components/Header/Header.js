import React from 'react';
import PropTypes from 'prop-types';
import './Header.scss';

import { Panel } from 'fundamental-react';

import SearchInput from './SearchInput/SearchInput';
import LabelsInput from './LabelsInput/LabelsInput';
import AdvancedSettings from './AdvancedSettings/AdvancedSettings';
import LabelsDisplay from './LabelsDisplay/LabelsDisplay';
import BottomToolbar from './BottomToolbar/BottomToolbar';

Header.propTypes = {
  updateFilteringState: PropTypes.func.isRequired,
  searchPhrase: PropTypes.string.isRequired,
  labels: PropTypes.array.isRequired,
  sortDirection: PropTypes.oneOf(['ascending', 'descending']),
  readonlyLabels: PropTypes.array.isRequired,
  advancedSettings: PropTypes.object.isRequired,
};

export default function Header({
  updateFilteringState,
  searchPhrase,
  labels,
  sortDirection,
  readonlyLabels,
  advancedSettings,
}) {
  const [advancedShown, setAdvancedShown] = React.useState(false);

  function addLabel(label) {
    if (!labels.includes(label)) {
      updateFilteringState({ labels: [...labels, label] });
    }
  }

  function removeLabel(label) {
    updateFilteringState({ labels: labels.filter(l => l !== label) });
  }

  function removeAllLabels() {
    updateFilteringState({ labels: [] });
  }

  function toggleAdvancedSettingsVisibility() {
    setAdvancedShown(!advancedShown);
  }

  const advancedSettingsButtonText = advancedShown
    ? 'Hide Advanced Settings'
    : 'Show Advanced Settings';

  return (
    <Panel className="fd-has-padding-regular fd-has-padding-bottom-none">
      <h1 className="fd-has-type-3 fd-has-padding-bottom-tiny">Logs</h1>
      <section className="header__settings-group">
        <LabelsInput readonlyLabels={readonlyLabels} addLabel={addLabel} />
        <SearchInput
          searchPhrase={searchPhrase}
          updateFilteringState={updateFilteringState}
        />
        <span
          className="link-button fd-has-type-minus-1 header__settings-group__toggle"
          onClick={toggleAdvancedSettingsVisibility}
        >
          {advancedSettingsButtonText}
        </span>
      </section>
      {advancedShown && (
        <AdvancedSettings
          advancedSettings={advancedSettings}
          hideSettings={() => setAdvancedShown(false)}
          updateFilteringState={updateFilteringState}
        />
      )}
      <div>
        <LabelsDisplay
          labels={labels}
          readonlyLabels={readonlyLabels}
          removeLabel={removeLabel}
          removeAll={removeAllLabels}
        />
        <BottomToolbar
          sortDirection={sortDirection}
          updateFilteringState={updateFilteringState}
        />
      </div>
    </Panel>
  );
}
