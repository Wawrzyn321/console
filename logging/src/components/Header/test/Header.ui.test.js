import React from 'react';
import { shallow } from 'enzyme';
import Header from './../Header';
import toJson from 'enzyme-to-json';

describe('Header UI', () => {
  it('Shows / hides advanced settings when corresponding button is clicked', () => {
    const component = shallow(
      <Header
        updateFilteringState={() => {}}
        searchPhrase="search phrase"
        labels={['a', 'b']}
        sortDirection="ascending"
        readonlyLabels={['c', 'd']}
        advancedSettings={{}}
      />,
    );

    const link = component
      .find('span[data-test-id="advanced-settings-toggle"]')
      .first();

    link.simulate('click');
    expect(toJson(component)).toMatchSnapshot();

    link.simulate('click');
    expect(toJson(component)).toMatchSnapshot();
  });
});
