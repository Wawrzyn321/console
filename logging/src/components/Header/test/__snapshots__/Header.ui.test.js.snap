// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header UI Shows / hides advanced settings when corresponding button is clicked 1`] = `
<Panel
  className="fd-has-padding-regular fd-has-padding-bottom-none"
>
  <h1
    className="fd-has-type-3 fd-has-padding-bottom-tiny"
  >
    Logs
  </h1>
  <section
    className="header__settings-group"
  >
    <LabelsInput
      addLabel={[Function]}
      maxRecentCount={4}
    />
    <SearchInput
      searchPhrase="search phrase"
      updateFilteringState={[Function]}
    />
    <span
      className="link-button fd-has-type-minus-1 header__settings-group__toggle"
      data-test-id="advanced-settings-toggle"
      onClick={[Function]}
    >
      Hide Advanced Settings
    </span>
  </section>
  <AdvancedSettings
    advancedSettings={Object {}}
    hideSettings={[Function]}
    updateFilteringState={[Function]}
  />
  <div>
    <LabelsDisplay
      labels={
        Array [
          "a",
          "b",
        ]
      }
      readonlyLabels={
        Array [
          "c",
          "d",
        ]
      }
      removeAll={[Function]}
      removeLabel={[Function]}
    />
    <BottomToolbar
      sortDirection="ascending"
      updateFilteringState={[Function]}
    />
  </div>
</Panel>
`;

exports[`Header UI Shows / hides advanced settings when corresponding button is clicked 2`] = `
<Panel
  className="fd-has-padding-regular fd-has-padding-bottom-none"
>
  <h1
    className="fd-has-type-3 fd-has-padding-bottom-tiny"
  >
    Logs
  </h1>
  <section
    className="header__settings-group"
  >
    <LabelsInput
      addLabel={[Function]}
      maxRecentCount={4}
    />
    <SearchInput
      searchPhrase="search phrase"
      updateFilteringState={[Function]}
    />
    <span
      className="link-button fd-has-type-minus-1 header__settings-group__toggle"
      data-test-id="advanced-settings-toggle"
      onClick={[Function]}
    >
      Hide Advanced Settings
    </span>
  </section>
  <AdvancedSettings
    advancedSettings={Object {}}
    hideSettings={[Function]}
    updateFilteringState={[Function]}
  />
  <div>
    <LabelsDisplay
      labels={
        Array [
          "a",
          "b",
        ]
      }
      readonlyLabels={
        Array [
          "c",
          "d",
        ]
      }
      removeAll={[Function]}
      removeLabel={[Function]}
    />
    <BottomToolbar
      sortDirection="ascending"
      updateFilteringState={[Function]}
    />
  </div>
</Panel>
`;
